### PR TITLE
Re-release the time based next/previous API

### PR DESCRIPTION
**Pull request by Exide**
**DONT MERGE**

I need to coordinate the merge order with some other PRs.

### CODE REVIEW

File: src/main/java/com/flightstats/hub/channel/LinkBuilder.java [hunk 1]
```diff
@@ -5,10 +5,7 @@
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import com.flightstats.hub.app.HubProvider;
 import com.flightstats.hub.metrics.ActiveTraces;
-import com.flightstats.hub.model.ChannelConfig;
-import com.flightstats.hub.model.ChannelContentKey;
-import com.flightstats.hub.model.ContentKey;
-import com.flightstats.hub.model.DirectionQuery;
+import com.flightstats.hub.model.*;
```
Comment (crozierm) [inline src/main/java/com/flightstats/hub/channel/LinkBuilder.java]: pet peave - can we avoid wildcard imports? I find explicit import helpful in PRs

File: src/main/java/com/flightstats/hub/channel/ChannelContentResource.java [hunk 2]
```diff
@@ -225,6 +226,81 @@ private Response getTimeQueryResponse(String channel, DateTime startTime, String
...
+                                         @QueryParam("epoch") @DefaultValue(Epoch.DEFAULT) String epoch,
+                                         @QueryParam("trace") @DefaultValue("false") boolean trace,
+                                         @QueryParam("stable") @DefaultValue("true") boolean stable,
+                                         @QueryParam("order") @DefaultValue(Order.DEFAULT) String order) {
+        SortedSet<ContentPath> keys = new TreeSet<>();
+        DateTime stableTime = TimeUtil.stable();
+        DateTime startTime = new DateTime(year, month, day, hour, minute, second, 999, DateTimeZone.UTC);
+        boolean next = direction.startsWith("n");
+        if (next) {
+            while (startTime.isBefore(stableTime) && keys.size() < count) {
```
Comment (crozierm) [inline src/main/java/com/flightstats/hub/channel/ChannelContentResource.java]: Am I reading this correctly: this just returns URLs to the time API, not links to the data items?

File: src/main/java/com/flightstats/hub/channel/LinkBuilder.java (hunk 1 above)
Comment (Exide) [reply src/main/java/com/flightstats/hub/channel/LinkBuilder.java]: Code I write doesn't use wildcard imports. I'm just trying to restore the master branch with the code that was there before the rollback. That said I'll go make the change.

Comment (crozierm) [reply src/main/java/com/flightstats/hub/channel/LinkBuilder.java]: I gotcha.

File: src/main/java/com/flightstats/hub/channel/ChannelContentResource.java (hunk 2 above)
Comment (Exide) [reply src/main/java/com/flightstats/hub/channel/ChannelContentResource.java]: Yup. This is an example payload:

```
{
  "_links" : {
    "self" : {
      "href" : "http://hub.dls.dev.flightstats.io/channel/single_test_1/2018/08/08/18/06/45/n/2"
    },
    "next" : {
      "href" : "http://hub.dls.dev.flightstats.io/channel/single_test_1/2018/08/08/18/06/46/next/2"
    },
    "previous" : {
      "href" : "http://hub.dls.dev.flightstats.io/channel/single_test_1/2018/08/08/18/06/45/previous/2"
    },
    "uris" : [
      "http://hub.dls.dev.flightstats.io/channel/single_test_1/2018/08/08/18/06/45",
      "http://hub.dls.dev.flightstats.io/channel/single_test_1/2018/08/08/18/06/46"
    ]
  }
}
```

**Pull request merged**